Fix warning about license-file of lib/cli/cardano-wallet-cli.cabal

rvl opened the pull request:
It was giving this warning:

```
Warning: 'license-file: ../../LICENSE' is a relative path outside of the
source tree. This will not work when generating a tarball with 'sdist'.
```

We could also simplify by just removing all `license-file` lines and the symlinks. This would be fine because the cabal files already say `license: MIT`, and there is a top-level `LICENSE` file.

uroboros commented on src/Cardano/Wallet/Kernel/DB/AcidState.hs:
```diff
@@ -548,26 +541,27 @@ createHdWallet newRoot defaultHdAccountId defaultHdAddress utxoByAccount =
+                                                   accountPublicKey
+                                                   addrPoolGap
+        , accountUpdateNew   = AccountUpdateNewUpToDate mempty
+        , accountUpdateAddrs = [] -- No addresses yet.
```
we need to add the new address pool addresses here (a new EOS Wallet-account must has gap number of HdAddrs)

denisshevchenko commented:
@KtorZ Conflicts (after merging of #256) are fixed.

KtorZ commented on src/Cardano/Wallet/Kernel/DB/HdWallet/Create.hs:
```diff
@@ -147,18 +147,20 @@ initHdRoot rootId name hasPass assurance created = HdRoot {
+    defName = AccountName $ sformat ("Account: " % build) accountIx
+    HdAccountId _ accountIx = case accountBase of
+        HdAccountBaseFO accId     -> accId
+        HdAccountBaseEO accId _ _ -> accId
```
Not sure we do want this (this = taking a HdAccountBase as an argument and doing a pattern matching inside the function). It'd make more sense to have separated functions for `initFOAccont` and `initEOAccount`. They'll likely take different parameters and, we don't want to mix them up in single functions. 

Thus, I'd keep the argument as it is right now, and only rename the function.

KtorZ commented on src/Cardano/Wallet/Kernel/DB/AcidState.hs:
```diff
@@ -707,14 +699,18 @@ accountUpdateCreate accId (AccountUpdateNewIncomplete curUtxo genUtxo ctx) =
   where
+    getAccountId updateBase = case updateBase of
+        AccountUpdateBaseFO accId     -> accId
+        AccountUpdateBaseEO accId _ _ -> accId
```
This (and all other occurrences of this) sounds like a good use-case for a lens. Actually, there's even already a function in `Kernel/DB/HdWallet` which does exactly that:

```hs
getHdAccountId :: HdAccountBase -> HdAccountId
getHdAccountId (HdAccountBaseFO accountId)     = accountId
getHdAccountId (HdAccountBaseEO accountId _ _) = accountId
```

This is likely something we'll keep on doing. So maybe having a lens here would be nice too (more consistent with the existing code-base). We already have:

```hs
hdAccountId :: Lens' HdAccount HdAccountId
hdAccountId = hdAccountBase . lens getHdAccountId setHdAccountId
```

So, maybe a `HdAccountBaseId` would do :+1:

rvl committed:
Fix warning about license-file of lib/cli/cardano-wallet-cli.cabal

Warning: 'license-file: ../../LICENSE' is a relative path outside of the
source tree. This will not work when generating a tarball with 'sdist'.
